c++/src/kj/async-io.c++

```diff
@@ -2764,6 +2764,121 @@ String CapabilityStreamNetworkAddress::toString() {
 
 // =======================================================================================
 
+namespace {
+
+class AggregateConnectionReceiver final: public ConnectionReceiver {
+public:
+  AggregateConnectionReceiver(Array<Own<ConnectionReceiver>> receiversParam)
```

Comment:
I guess if someone for some reason passed an empty array of receivers, the `accept` methods would still work correctly, in the sense that they'd return promises that never resolve.  But it looks like the couple uses of `receivers[0]` are only guarded by an `KJ_IREQUIRE`, so maybe we should require a non-empty array, either in those methods or in the constructor?

Reply:
Yeah, we should probably KJ_REQUIRE that the array is non-empty...